cookbook/.cursor/rules/cookbook-recipe-subscriptions.mdc

```diff
@@ -157,6 +158,310 @@ export function SellingPlanSelector({
+  }
+
+  const form = await request.formData();
+
+  try {
+    const subId = form.get('subId');
+
+    if (!subId) {
+      throw new Error('Subscription ID is required');
+    }
+
+    await customerAccount.mutate(SUBSCRIPTION_CANCEL_MUTATION, {
+      variables: {
+        subscriptionContractId: subId.toString(),
+      },
+    });
+
+    return {
+      error: null,
+    };
+  } catch (error: any) {
+    return data(
+      {
+        error: error.message,
+      },
+      {
+        status: 400,
+      },
+    );
+  }
+}
+
+export default function AccountProfile() {
+  const {state} = useNavigation();
+  const action = useActionData<ActionResponse>();
+
+  const {subscriptions} = useLoaderData<typeof loader>();
+
+  const sortedSubscriptions =
+    subscriptions?.customer?.subscriptionContracts.nodes.sort((a, b) => {
+      // active subscriptions first
+      if (a.status === 'ACTIVE' && b.status !== 'ACTIVE') {
+        return -1;
+      }
+      if (a.status !== 'ACTIVE' && b.status === 'ACTIVE') {
+        return 1;
+      }
+      // then, newest first
+      return new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime();
+    });
+
+  return (
+    <div className="account-profile">
+      <h2>My subscriptions</h2>
+      {action?.error ? (
+        <p>
+          <mark>
+            <small>{action.error}</small>
+          </mark>
+        </p>
+      ) : null}
+      <div className="account-subscriptions">
+        {sortedSubscriptions.map((subscription) => {
+          return (
+            <div key={subscription.id} className="subscription-row">
+              <div className="subscription-row-content">
+                <div>
+                  {subscription.lines.nodes.map((line) => (
+                    <div key={line.id}>{line.name}</div>
+                  ))}
+                </div>
+                <div>
+                  Every{' '}
+                  <SubscriptionInterval
+                    billingPolicy={subscription.billingPolicy}
+                  />
+                </div>
+              </div>
+              <div className="subscription-row-actions">
+                <div
+                  className={
+                    subscription.status === 'ACTIVE'
+                      ? 'subscription-status-active'
+                      : 'subscription-status-inactive'
+                  }
+                >
+                  {subscription.status}
+                </div>
+                {subscription.status === 'ACTIVE' && (
+                  <Form key={subscription.id} method="DELETE">
+                    <input
+                      type="hidden"
+                      id="subId"
+                      name="subId"
+                      value={subscription.id}
+                    />
+                    <button type="submit" disabled={state !== 'idle'}>
+                      {state !== 'idle' ? 'Canceling' : 'Cancel subscription'}
+                    </button>
```

Comment:
Would this mean that multiple subscriptions, canceling one, would mean that the button for canceling another would also say "Canceling" briefly?